File: Sistema_de_Gestión_de_Productos.java
class Producto{
    private String nombre;
    private int precio;
    private String categoria;

    public Producto(String nombre, int precio, String categoria){
        this.nombre = nombre;
        this.precio = precio;
        this.categoria = categoria;
    }

    public void mostrar_detalle(){
        System.out.println("\nNombre: "+nombre+"\nPrecio: "+precio+"\nCategoria: "+categoria);
    }
}

class Electrónico extends Producto{
    private String marca;
    private String modelo;

    public Electrónico(String nombre, int precio, String categoria, String marca, String modelo){
        super(nombre, precio, categoria);
        this.marca = marca;
        this.modelo = modelo;
    }

    public void mostrar_detalle(){
        super.mostrar_detalle();
        System.out.println("Marca del Electrónico: "+marca+"\nModelo del electrónico: "+modelo);
    }
}

class Alimenticio extends Producto{
    private String marca;
    private String fecha_de_vencimiento;

    public Alimenticio(String nombre, int precio, String categoria, String marca, String fecha_de_vencimiento){
        super(nombre, precio, categoria);
        this.marca = marca;
        this.fecha_de_vencimiento = fecha_de_vencimiento;
    }

    public void mostrar_detalle(){
        super.mostrar_detalle();
        System.out.println("Marca del Alimento: "+marca+"\nFecha de vencimiento del alimento: "+fecha_de_vencimiento);
    }
}

class Vestimenta extends Producto{
    private String marca;
    private String tallas;

    public Vestimenta(String nombre, int precio, String categoria, String marca, String tallas){
        super(nombre, precio, categoria);
        this.marca = marca;
        this.tallas = tallas;
    }

    public void mostrar_detalle(){
        super.mostrar_detalle();
        System.out.println("Marca: "+marca+"\nTallas disponibles: "+tallas);
    }
}



public class Sistema_de_Gestión_de_Productos{
    public static void main(String args[]){
        Electrónico electrónico1 = new Electrónico("Celular", 199000, "Dispositivos móviles", "LG", "G7");
        electrónico1.mostrar_detalle();
        Alimenticio alimento1 = new Alimenticio("Porotos", 2990, "Legumbres", "Wasil", "17/12/2023");
        alimento1.mostrar_detalle();
        Vestimenta vestimenta1 = new Vestimenta("Zapatilla", 79000, "Calzado", "Nike SB", "39, 41, 42, 43, 44");
        vestimenta1.mostrar_detalle();
    }
}

File: Sistema de Gestión de Productos.py
class Producto:
    def __init__(self, nombre, precio, categoria):
        self.nombre = nombre
        self.precio = precio
        self.categoria = categoria
    
    def mostrar_detalle(self):
        print(f"\n{self.nombre}:\nSu precio es de ${self.precio}.\nPertenece a la categoria {self.categoria}.")

class Electrónico(Producto):
    def __init__(self, nombre, precio, categoria, marca, modelo):
        super().__init__(nombre, precio, categoria)
        self.marca = marca
        self.modelo = modelo

    def mostrar_detalle(self):
        super().mostrar_detalle()
        print(f"La marca del {self.nombre} es {self.marca}.\nEl modelo del {self.nombre} es {self.modelo}.")

class Alimenticio(Producto):
    def __init__(self, nombre, precio, categoria, marca, salud):
        super().__init__(nombre, precio, categoria)
        self.marca = marca
        self.salud = salud

    def mostrar_detalle(self):
        super().mostrar_detalle()
        print(f"La marca del {self.nombre} es {self.marca}.\nEs {self.salud} para el organismo.")

class Vestimenta(Producto):
    def __init__(self, nombre, precio, categoria, marcamodelo, talla):
        super().__init__(nombre, precio, categoria)
        self.marcamodelo = marcamodelo
        self.talla = talla

    def mostrar_detalle(self):
        super().mostrar_detalle()
        print(f"La marca y el modelo son: {self.marcamodelo}.\nLas tallas disponibles son: {self.talla}.")


Celular = Electrónico("Celular", 499900, "Electrónico", "Samsung", "Galaxy Note 20 PRO")
Celular.mostrar_detalle()
Notebook = Electrónico("Notebook", 599000, "Electrónico", "Asus", "Tuf Gaming F15")
Notebook.mostrar_detalle()
Carne = Alimenticio("Lomo Vetado", 12990, "Alimenticio", "Corte Criollo", "Saludable")
Carne.mostrar_detalle()
Torta = Alimenticio("Torta", 5990, "Alimenticio", "Tortas Marcela", " Dañino")
Torta.mostrar_detalle()
Zapatilla = Vestimenta("Zapatilla", 59990, "Vestimenta", "Reebok Court Peak", "40, 41, 42, 43, 44, 45")
Zapatilla.mostrar_detalle()
Sudadera = Vestimenta("Sudadera", 19990, "Vestimenta", "Oneill HODDIE SUNRISE Gris", "S, M, L, XL, XXL")
Sudadera.mostrar_detalle()